2015/presentations/how_to_re/text.md

```diff
@@ -0,0 +1,43 @@
+Zainspirowani pracą Turinga, dwaj neurolodzy Warren McCulloch i Walter Pitts zbudowali sztuczny neuron, który jest uproszczonym matematycznym modelem biologicznego neuronu. Służy jako podstawowy blok budulcowy sztucznych sieci neuronowych. [4] W 1943 wyniki swojej pracy opublikowali w artykule naukowym w Bulletin of Mathematical Biophysics 2:115-133 zatytułowanym: "A logical calculus of the ideas immanent in nervous activity". Okazało się, że ten artykuł stał się podwaliną pod rozwój współczesnej informatyki, chociaż nie taki był jego cel. [5]
+
+### b) Co tam panie, w matematyce słychać ###
+Kilkanaście lat później, w 1956 roku, matematyk Stephen Kleene poszedł o krok dalej i bazując na neuronie McCullocha-Pittsa [4], zaproponował prostą algebrę, swoje przemyślenia publikując w artykule "Representation of events in nerve nets and finite automata". [6] Jego celem było opisanie mózgu jako rachunku logicznego.
+Kleene'a nie interesowało to, czy model McCullocha-Pittsa był dokładny; skupił się na tym czym ten model jest. W przeciwieństwie do podejścia McCullocha-Pittsa, Kleene udowodnił, że model neuronu nie ogranicza się jedynie do ludzkiego mózgu, ale w rzeczywistości opisuje szerszy zakres jakim jest każdy automat skończony, bez ograniczeń na to, czy jest to zwierzę, człowiek czy maszyna. Tym samym, nawiązał do wykładu o automatach, jaki von Neumann dał w trakcie Sympozjum Hixona. [9] To właśnie John von Neumann był jednym z pierwszych, którzy próbowali przedstawić ogólną teorię automatów wykorzystując narzędzia logiki.
+Artykuł Kleene'a był pierwszym krokiem na drodze ku temu, by można było wyrażenia regularne zaszczepić w programach komputerowych.
+
+### c) Jak by tu tego użyć w informatyce ###
+Prawdopodobnie efekt pracy, którą dokonali wymienieni wcześniej naukowcy, nigdy by nie trafił do komputerów, gdyby nie Janusz Brzozowski. [10] Brzozowski, który uzyskał tytuł doktora nauk elektrycznych na Princeton University, przez pewien czas wykładał w Berkeley. Wtedy też zaprezentował sposób na wykorzystanie wyrażeń regularnych by stworzyć diagram stanów, które były szeroko wykorzystywane w latach 50-tych do projektowania i implementowania programów w ówczesnych komputerach.
+Właśnie w Berkeley, Ken Thompson [12] miał z Brzozowskim do czynienia. Bazując na tym co usłyszał, postanowił zaadoptować to podejście, i zaimplemenować je w oprogramowaniu. Pracując przy tworzeniu UNIX-a, napisał coś, co ostatecznie miało stać się narzędziem "grep" a zaimplementowane było w edytorze "QED". "grep" to w rzeczywistości skrót od "Globally search for regular expression re and Print it", czyli w wolnym tłumaczeniu "globalne wyszukiwanie wyrażeń regularnych oraz wyświetlanie ich". Swój algorytm opisał w artykule "Regular Expression Search Algorithm" [14], a całość zaimplementował w asemblerze dla IBM 7090.
```

Comment:
Dość dziwnie brzmi to zdanie: "Właśnie w Berkeley, Ken Thompson [12] miał z Brzozowskim do czynienia", może lepiej: "Właśnie w Berkeley, z Brzozowskim zetknął się Ken Thompson [12]"

zaadoptować -> zaadaptować